File: Tcf.cpp
#include <iostream>
#include <string>
using namespace std;

// Thanks to kobok on sorcerynet irc for providing a proper c++ example.
 
class Finally {
  string msg;
  public:
  Finally(string msg) : msg(msg) { }
  ~Finally() { cout << msg; }
};
 
int main() {
    Finally f("Outer Finally\n");
    try {
        Finally f("Inner Finally\n");
        try {
            throw "Foo";
        }
        catch (const char* e) {
            cout << "Inner Catch\n";
            throw e;
        }
    }
    catch (const char* e) {
        cout << "Outer Catch\n";
    }
    return 0;
}

File: Tcf.js
try {
	try {
		throw new Error("Foo");
	}
	catch (ex) {
		print("Inner Catch");
		throw ex;
	}
	finally {
		print("Inner Finally");
	}
}
catch (ex) {
	print("Outter Catch");
}
finally {
	print("Outter Finally");
}